### ISSUE TITLE
Fix request: pmd7 OptimizeMapOrSetForEnum false positive and missed case

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#387]
```
class Issue387 {
    private enum Language {
    }
    private static final Map<Language, String> RULES_NAMES;

    static {
        Map<Language, String> map = new EnumMap<>(Language.class);
        RULES_NAMES = Collections.unmodifiableMap(map); // <--- false positive
    }
    private static final Map<ChronoField, String> mapEnumValueInLine = Map.of( // bad, missed
                ChronoField.DAY_OF_WEEK, "DOW",
                ChronoField.DAY_OF_MONTH, "DOM"
    );
    private static final Map<ChronoField, String> mapEnumValueInLineOneElem = Map.of( // good, 1 entry no hashmap overhead
                ChronoField.DAY_OF_WEEK, "DOW"
    );
}
```
